Content.Server/Radiation/Components/RadiationProtectionComponent.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Robust.Shared.Prototypes;
+using Content.Shared.Damage.Prototypes;
+
+namespace Content.Server.Radiation.Components;
+
+[RegisterComponent]
+public sealed partial class RadiationProtectionComponent : Component
+{
+    /// <summary>
+    ///     The radiation damage modifier for entities with this component.
+    /// </summary>
+    [DataField("modifier")]
+    public ProtoId<DamageModifierSetPrototype> RadiationProtectionModifierSetId = "PotassiumIodide";
+}
```

Comment:
think this could very easily be genericized to just like a `DamageProtectionBuffComponent`, its not really specific to radiation except that the default field is one for rads

Reply:
I like the idea but the problem here is its use as a status effect. StatusEffectsSystem.cs adds this component and removes it after a certain time is up. For example, if we had an radiation protection status effect and a seperate electrocution protection status effect, the component for both would get removed when the time for one is up. So right now each status effect requires its own component.

Reply:
I was trying to create DamageProtectionBuffComponent and make RadiationProtectionComponent just inherit from it, overwrite the ModifierSetId and have a separate component for the status effect that way, but it seems like component inheritance is not a thing in this codebase? The status effect system might need some changes to prevent component spam.